src/main/java/de/unistuttgart/overworldbackend/service/PlayerTaskStatisticService.java

```diff
@@ -134,30 +140,27 @@ public PlayerTaskStatisticDTO submitData(final PlayerTaskStatisticData data) {
     );
   }
 
-  private void calculateCompletedDungeons(final PlayerStatistic playerstatistic) {
+  private void isDungeonCompleted(Dungeon dungeon, PlayerStatistic playerStatistic) {
     final List<PlayerTaskStatistic> playerTaskStatistics = playerTaskStatisticRepository.findByPlayerStatisticId(
-      playerstatistic.getId()
-    );
-    playerstatistic.setCompletedDungeons(
-      playerstatistic
-        .getCourse()
-        .getWorlds()
-        .parallelStream()
-        .map(World::getDungeons)
-        .flatMap(Collection::parallelStream)
-        .filter(dungeon ->
-          dungeon
-            .getMinigameTasks()
-            .parallelStream()
-            .allMatch(minigameTask ->
-              playerTaskStatistics
-                .parallelStream()
-                .filter(taskStatistic -> taskStatistic.getMinigameTask().equals(minigameTask))
-                .anyMatch(PlayerTaskStatistic::isCompleted)
-            )
-        )
-        .collect(Collectors.toCollection(ArrayList::new))
+      playerStatistic.getId()
     );
+    long currentTime = System.nanoTime();
+    boolean dungeonCompleted = dungeon
+      .getMinigameTasks()
+      .parallelStream()
+      .allMatch(minigameTask ->
+        playerTaskStatistics
+          .parallelStream()
+          .filter(playerTaskStatistic -> playerTaskStatistic.getMinigameTask().equals(minigameTask))
+          .anyMatch(PlayerTaskStatistic::isCompleted)
+      );
+    long duration = System.nanoTime() - currentTime;
+    System.out.println(duration);
```

Comment:
Maybe annotate the class with `@Slf4j` and use log.debug(). Is this logging needed in the future?

Reply:
1. What's the reason for the duration?
2. Why in nano seconds?